pkg/crc/network/proxy.go

```diff
@@ -0,0 +1,143 @@
+package network
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+)
+
+var defaultNoProxies = []string{".apps-crc.testing", "127.0.0.1", "api.crc.testing", "localhost"}
+
+// ProxyConfig keeps the proxy configuration for the current environment
+type ProxyConfig struct {
+	httpProxy  string
+	httpsProxy string
+	noProxy    []string
+}
+
+// NewProxyConfig creates a proxy configuration with the specified parameters. If a empty string is passed
+// the corresponding environment variable is checked.
+func NewProxyConfig(httpProxy string, httpsProxy string, noProxy string) (*ProxyConfig, error) {
+	defaultScheme := "http"
+
+	if httpProxy == "" {
+		httpProxy = getProxyFromEnv("http_proxy")
+	}
+	err := ValidateProxyURL(httpProxy, defaultScheme)
+	if err != nil {
+		return nil, err
+	}
+
+	if httpsProxy == "" {
+		httpsProxy = getProxyFromEnv("https_proxy")
+	}
+	err = ValidateProxyURL(httpsProxy, defaultScheme)
+	if err != nil {
+		return nil, err
+	}
+
+	np := []string{}
+	np = append(np, defaultNoProxies...)
+
+	if noProxy == "" {
+		noProxy = getProxyFromEnv("no_proxy")
+	}
+
+	if noProxy != "" {
+		np = append(np, strings.Split(noProxy, ",")...)
+	}
+
+	config := ProxyConfig{
+		httpProxy:  httpProxy,
+		httpsProxy: httpsProxy,
+		noProxy:    np,
+	}
+
+	return &config, nil
+}
+
+func getProxyFromEnv(proxyScheme string) string {
+	p := os.Getenv(proxyScheme)
```

Comment:
@cfergeau these can contain proxy authentication... so when a separate option is used, how to reliably recontruct or handle this?

Reply:
Context seems to be missing, I assume this was an answer to my comment saying we can have separate options if we want the user to be able to pass unescaped passwords/usernames?
We have to define priority rules between env and config anyways, so we'll just say "anything defined in the config overrides anything set in the environment" (or the opposite). This is not much different from setting both http_proxy and the corresponding configuration option.

Reply:
@cfergeau As of now anything which is part of config should overwrite what's in the ENV.

> With that said, I dont' think we need dedicated username/password options just now, this is just something to keep in mind if user feedback tells us it's needed.

Yes, agreed 👍